src/validation.cpp

```diff
@@ -3889,6 +3896,13 @@ bool BlockManager::LoadBlockIndex(
         }
         if (ShutdownRequested()) return false;
         CBlockIndex* pindex = item.second;
+
+        if (pindex->pprev) {
+            for (unsigned i = 0; i < NUM_ALGOS_IMPL; i++)
+              pindex->lastAlgoBlocks[i] = pindex->pprev->lastAlgoBlocks[i];
```

Comment:
also useless - only null ptr in the array for the other algos

Reply:
@otherdeniz We don't claim to have perfect code for tweaking other people's software. We came up with this quick solution ourselves, we used it for our own needs, and it works for us. We shared it with developers. Developers can, if they wish, understand the idea and rewrite this code, taking into account their understanding of program structure and acceptable methods.